extension/lib/display.js

```diff
@@ -0,0 +1,128 @@
+        this.time_remaining = this.initTimer(readableDomIds);
+        this.reading_speed = this.initSpeed(speed);
+        this.end = null; 
+
+        this.defineHtml();
+        this.createDisplay(readableDomIds);
+    }
+
+    /*
+    Defines html code for display to be inserted into web page. 
+    */
+    defineHtml() {
+        this.html = `
+            <div id="displayContainer">
+                <div id="timerContainer">
+                    <div id="timerInsideContainer">
+                        <span id="timerNumber">${this.time_remaining}</span> min remaining
+                    </div>
+                </div>
+                <div id="speedContainer">
+                    <div id="speedInsideContainer">
+                        <span id="speedNumber">${this.reading_speed}</span> WPM
+                    </div>
+                </div>
+            </div>
+            `;      
+    }
+
+    /*
+    Inserts display HTML into webpage. 
+    */
+    createDisplay(readableDomIds) {
+        document.getElementById(readableDomIds[0]).innerHTML = this.html;
+        document.getElementById("displayContainer").style.opacity = 1; // For smoother transition
+    }
+
+    /* 
+    Params: 
+    - List of container IDs on web page (string[])
+    Return: 
+    The total time remaining to read document, based on avg letters/word and lowball estimate of WPM (int)
+    */
+    initTimer(readableDomIds) {
+        let total_char = 0;
+        let avg_letters_per_word = 6; // Actual is 4.79 (from Quora), add a space after each word to get 6. 
+        let avg_read_speed = 200; // Low ball estimate, from Rayner (in WPM)
+        for (var section in readableDomIds) { // Get total_char of all containers in web page
+            let elem = document.getElementById(readableDomIds[section]);
+            let text_len = elem.innerText.length;
+            total_char = total_char + text_len;
+        }
+        let total_words = total_char/avg_letters_per_word;
+        let time_remaining = total_words/avg_read_speed; // in minutes
+        return Math.ceil(time_remaining);
+    }
+    
+    /*
+    Updates reading timer based on containers after current tracker. 
+    */
+    updateTimer(readableDomIds, containerId) { // Call everytime you get to new paragraph
+        let total_char = 0;
+        for (var section in readableDomIds.slice(containerId)) {
+            let elem = document.getElementById(readableDomIds[section]);
+            let text_len = elem.innerText.length;
+            total_char = total_char + text_len;
+        }
+        let total_words = total_char/avg_letters_per_word;
+        let time_remaining = total_words/avg_read_speed; // in minutes
+        this.time_remaining = Math.ceil(time_remaining);
+        document.getElementById("timerNumber").innerHTML = this.time_remaining;
+    }
+
+    /*
+    Returns: 
+    - Initial speed in WPM, based on speed parameter used by tracker (int)
+    */
+    initSpeed(speed) {
```

Comment:
Two methods below are good for now.
Can you leave a TODO: to show that it's not final?